test/logging_test.cc

```diff
@@ -0,0 +1,51 @@
+#include "ex_actor/internal/logging.h"
+
+#include <memory>
+#include <sstream>
+
+#include <gtest/gtest.h>
+#include <spdlog/sinks/ostream_sink.h>
+#include <spdlog/spdlog.h>
+
+#include "ex_actor/api.h"
+
+TEST(LoggingTest, SetLogLevelTest) {
+  // default log level - should see output
+  {
+    std::ostringstream oss;
+    auto test_sink = std::make_shared<spdlog::sinks::ostream_sink_mt>(oss);
+
+    // Set the test sink before Init
+    ex_actor::internal::logging::GetTestSink() = test_sink;
+
+    ex_actor::Init(/*thread_pool_size=*/10);
+    ex_actor::Shutdown();
+
+    std::string output = oss.str();
+
+    // Clear the test sink
+    ex_actor::internal::logging::GetTestSink().reset();
+
+    ASSERT_TRUE(!output.empty()) << "Expected log output but got none";
+  }
+
+  // suppressed log level - should not see output
+  {
+    std::ostringstream oss;
+    auto test_sink = std::make_shared<spdlog::sinks::ostream_sink_mt>(oss);
+
+    // Set the test sink before Init
+    ex_actor::internal::logging::GetTestSink() = test_sink;
+
+    ex_actor::Init(/*thread_pool_size=*/10);
+    ex_actor::SetLogLevel(spdlog::level::warn);
```

Comment:
Maybe add a new case to cover dynamic log level switching ? For example, set to debug and output a log, then set to info and ensure debug logs are no longer output ?